lib/Backend/IR.h

```diff
@@ -176,10 +176,10 @@ class Instr
     {
     }
 public:
-    static Instr *  New(Js::OpCode opcode, Func *func);
```

Comment:
Is it true that most calls to Instr::New will take a null pointer for byteCodeOffsetInstr? Would it make sense to have a new API for the few places that have a non-null byteCodeOffsetInstr, to save the extra parameter?

Reply:
Over time, I have noticed multiple instances where we call `SetByteCodeOffset` on an instr immediately after creating it. I didn't visit all instances of calling `Instr::New` to see if they can be simplified by passing the extra param, but I thought it could ease things in the future.

Reply:
Most calls to IR::Instr::New don't do that. The common idiom in IRBuilder is to call AddInstr, does call the uint overload directly. (I guess it's allowed to.) And later ones (Lower, e.g.) tend to leave m_offset at -1.

Reply:
Ok. I'll revert the change to Instr::New and its overloads. Creating an overload that takes an optional uint parameter and cleaning up IRBuilder is out of scope for this PR. Would be good if we do it eventually.

Reply:
Are you sure? I was thinking of an overload that takes a (non-optional) Instr*. That's pretty trivial, and it would let you keep 99% of your change as is.

Reply:
Oh, I misunderstood you. Yeah, that works. Updated.

Reply:
Is this the updated version? Is the optional Instr* going away?

Reply:
Yes, the latest version doesn't have the optional Instr*.

Reply:
Oh, I see it. Cool. Thanks.